### PR TITLE
added infra with local k8s, helm and minikube

### PR DESCRIPTION
**Pull request by BatiGencho**
This MR adds full infra for local dev and also k8s deployment (on cloud) with k8s, Helm and minikube. I have also added Tilt as a local dev tool.

### CODE REVIEW

File: k8s/minikube/crds/traefik-resources.yaml [hunk 1]
```diff
@@ -0,0 +1,1598 @@
+---
```
Comment (arminsabouri) [inline k8s/minikube/crds/traefik-resources.yaml:1]: Just a question. Im not too familiar with minikube. How was this file generated? Does it need to be updated as we update our minikube setup?

File: k8s/minikube/files/master/monitoring/grafana/dashboards/testnet.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (arminsabouri) [inline k8s/minikube/files/master/monitoring/grafana/dashboards/testnet.json]: Are we able to view the grafana dashboards generated via Reth?
https://paradigmxyz.github.io/reth/run/observability.html

File: k8s/minikube/install_minikube.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+[[ $DEBUG = true ]] && set -x
+set -euo pipefail
+
+minikube delete && \
+sudo rm -rf /usr/local/bin/minikube && \
+sudo curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && \
```
Comment (arminsabouri) [inline k8s/minikube/install_minikube.sh:8]: Would this fail for the devs using the new arm arch machines?
If so we could just have different versions of this script

File: k8s/minikube/templates/btc-server.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    matchLabels:
+      service: btcserver
+  template:
+    metadata:
+      labels:
+        service: btcserver
+    spec:
+      #serviceAccountName: {{ .Values.serviceAccount }}
+      #imagePullSecrets:
+      #- name: botanix-full-stack
```
Comment (arminsabouri) [inline k8s/minikube/templates/btc-server.yaml]: Can we remove this?

File: k8s/minikube/templates/elasticsearch.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    matchLabels:
+      service: elasticsearch
+  template:
+    metadata:
+      labels:
+        service: elasticsearch
+    spec:
+      serviceAccountName: {{ .Values.serviceAccount }}
+      #imagePullSecrets:
+      #- name: botanix-full-stack
```
Comment (arminsabouri) [inline k8s/minikube/templates/elasticsearch.yaml]: Same thing here. Can we remove this?

File: k8s/minikube/templates/kibana.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        service: kibana
+    spec:
+      serviceAccountName: {{ .Values.serviceAccount }}
+      #imagePullSecrets:
+      #- name: botanix-full-stack
```
Comment (arminsabouri) [inline k8s/minikube/templates/kibana.yaml]: Same thing can we remove this?

File: k8s/minikube/templates/sidecar.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  creationTimestamp: null
+  labels:
+    service: sidecar
+  name: sidecar
+spec:
+  ports:
+    - name: grpc
+      port: 50051
+      protocol: TCP
+      targetPort: 50051
```
Comment (arminsabouri) [inline k8s/minikube/templates/sidecar.yaml]: how is this port `50051` being used?

File: k8s/minikube/templates/sidecar.yaml [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    matchLabels:
+      service: sidecar
+  template:
+    metadata:
+      labels:
+        service: sidecar
+    spec:
+      #serviceAccountName: {{ .Values.serviceAccount }}
+      #imagePullSecrets:
+      #- name: seatlab-full-stack
```
Comment (arminsabouri) [inline k8s/minikube/templates/sidecar.yaml]: Can we remove?

File: k8s/minikube/templates/sidecar.yaml [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      - env:
+        - name: HOST
+          value: "0.0.0.0"
+        - name: PORT
+          value: "50051"
+        name: sidecar
+        imagePullPolicy: Always
+        image: "{{ .Values.sidecar.image }}:{{ .Values.sidecar.tag }}"
+        ports:
+          - containerPort: 50051
```
Comment (arminsabouri) [inline k8s/minikube/templates/sidecar.yaml]: IIRC default port for sidecar is 8081

File: k8s/minikube/templates/testnet.yaml [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  creationTimestamp: null
+  labels:
+    service: testnet
+  name: testnet
+spec:
+  ports:
+    - name: port-30303
```
Comment (arminsabouri) [inline k8s/minikube/templates/testnet.yaml]: Nit: Would it be beneficial to name these ports based on their purpose?
i.e `name: reth-p2p`, `name: reth-metrics`

**Review by arminsabouri — COMMENTED** [review]
Thank you for taking the time to set all this up!
I had a couple nits and questions. Most of my questions are to help me better understand this setup, not necessarily requests for change.

I am going to take sometime to set this up before I review.
@scottmillner when you get a chance can you also take a look

**Comment by scottmillner** [comment]
> Thank you for taking the time to set all this up! I had a couple nits and questions. Most of my questions are to help me better understand this setup, not necessarily requests for change.
> 
> I am going to take sometime to set this up before I review. @scottmillner when you get a chance can you also take a look

~~i'm having a few issues running everything when following the readme~~
might be nice to have Evgeni do a walk through with us

i got everything running

File: k8s/minikube/install_minikube.sh (hunk 3 above)
Comment (scottmillner) [reply k8s/minikube/install_minikube.sh:8]: yeah...i just used brew to install minikube for my M1

File: k8s/minikube/templates/testnet.yaml (hunk 10 above)
Comment (BatiGencho) [reply k8s/minikube/templates/testnet.yaml]: done

File: k8s/minikube/templates/sidecar.yaml (hunk 9 above)
Comment (BatiGencho) [reply k8s/minikube/templates/sidecar.yaml]: my bad, sorry!.Done

File: k8s/minikube/templates/sidecar.yaml (hunk 8 above)
Comment (BatiGencho) [reply k8s/minikube/templates/sidecar.yaml]: ofc, done

File: k8s/minikube/templates/sidecar.yaml (hunk 7 above)
Comment (BatiGencho) [reply k8s/minikube/templates/sidecar.yaml]: the port was wrong, I fixed it with the correct one. My bad, sorry

File: k8s/minikube/templates/kibana.yaml (hunk 6 above)
Comment (BatiGencho) [reply k8s/minikube/templates/kibana.yaml]: ofc, removed

File: k8s/minikube/templates/elasticsearch.yaml (hunk 5 above)
Comment (BatiGencho) [reply k8s/minikube/templates/elasticsearch.yaml]: ofc, removed

File: k8s/minikube/templates/btc-server.yaml (hunk 4 above)
Comment (BatiGencho) [reply k8s/minikube/templates/btc-server.yaml]: ofc, removed

File: k8s/minikube/install_minikube.sh (hunk 3 above)
Comment (BatiGencho) [reply k8s/minikube/install_minikube.sh:8]: @scottmillner do you mind adding what you did for MacOS to that sh (or maybe adding a new .sh for macos)

File: k8s/minikube/files/master/monitoring/grafana/dashboards/testnet.json (hunk 2 above)
Comment (BatiGencho) [reply k8s/minikube/files/master/monitoring/grafana/dashboards/testnet.json]: yes, I have added the dashboard from the official reth repo. Its called overview.json

File: k8s/minikube/crds/traefik-resources.yaml (hunk 1 above)
Comment (BatiGencho) [reply k8s/minikube/crds/traefik-resources.yaml:1]: Traeffic is a loadbalancer and it does many other things too. We are not really using it here. It does not need to be updated as it is completely independent from the services we are using. We can keep it for now I guess, once we have our devops, he could decide if we need it for cluster deployment or not. Just ignore it for now.

**Review by scottmillner — APPROVED** [review]
💪 🔥 🔥

**Comment by BatiGencho** [comment]
@birozuru @0xBEEFCAF3  please review again

File: k8s/minikube/files/master/monitoring/prometheus.yaml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+alerting:
+  alertmanagers:
+  - static_configs:
+    - targets:
+      # - alertmanager:9093
+
+# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
+rule_files:
+  # - "first_rules.yml"
+  # - "second_rules.yml"
```
Comment (arminsabouri) [inline k8s/minikube/files/master/monitoring/prometheus.yaml]: Nit: Can we remove some of the comments here?

**Review by arminsabouri — APPROVED** [review]
Good to go!

File: k8s/minikube/files/master/monitoring/prometheus.yaml (hunk 11 above)
Comment (BatiGencho) [reply k8s/minikube/files/master/monitoring/prometheus.yaml]: sure, done

**Pull request merged**